### PR TITLE
chore: regenerate parser

### LINKED ISSUE #2: feature: Regenerate bindings for latest version of tree-sitter
### Did you check the tree-sitter docs?

- [X] I have read all the [tree-sitter docs](https://tree-sitter.github.io/tree-sitter/using-parsers) if it relates to using the parser

### Is your feature request related to a problem? Please describe.

Regenerate bindings for latest version of tree-sitter

Publish them to PYPI, npm, crates.io

### Describe the solution you'd like

Use latest tree-sitter rerun `tree-sitter generate`.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Freed-Wu**
Fix #2

What is your PYPI user name? I create https://pypi.org/project/tree-sitter-make/ and need to add you :smile:

### CODE REVIEW

**Comment by amaanq** [comment]
I'm not sure why *you* would make the pypi package for a grammar that is *not* your own, that doesn't make any sense. The contribution to update the bindings is very welcome, but you are not a member of the org nor is this your grammar to begin with, nor are you a maintainer of it. I'd ask you to delete the package and let one of us in control of the publishing here take care of that. Please do not *ever* publish other people's packages for them, I've had to deal with frustrating experiences stemming from this kind of action [before](https://github.com/tree-sitter/tree-sitter-html/issues/68#issuecomment-1846130003), and it is very annoying to say the least. Once deleted, I can go ahead and get this in + publish this to the appropriate registries

**Comment by Freed-Wu** [comment]
I have added you to owners. Before you accepted, I can stop any others to use this name. After accepted, Just remove me is OK.

**Comment by amaanq** [comment]
Alright - thanks for that, but in general always *ask* before squatting, it is really frustrating and a no-no, I'll review this now

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[package]
+name = "tree-sitter-make"
+description = "Make grammar for tree-sitter"
+version = "0.0.1"
+license = "MIT"
+readme = "README.md"
+keywords = ["incremental", "parsing", "tree-sitter", "make"]
+categories = ["parsing", "text-editors"]
+repository = "https://github.com/tree-sitter/tree-sitter-make"
```
Comment (amaanq) [inline Cargo.toml]:
```suggestion
repository = "https://github.com/tree-sitter-grammars/tree-sitter-make"
```

File: Package.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                    "examples",
+                    ".editorconfig",
+                    ".github",
+                    ".gitignore",
+                    ".gitattributes",
+                    ".gitmodules",
+                ],
+                sources: [
+                    "src/parser.c",
+                    // NOTE: if your language has an external scanner, add it here.
```
Comment (amaanq) [inline Package.swift:38]:
```suggestion
```

File: binding.gyp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      "dependencies": [
+        "<!(node -p \"require('node-addon-api').targets\"):node_addon_api_except",
+      ],
+      "include_dirs": [
+        "src",
+      ],
+      "sources": [
+        "bindings/node/binding.cc",
+        "src/parser.c",
+        # NOTE: if your language has an external scanner, add it here.
```
Comment (amaanq) [inline binding.gyp:14]:
```suggestion
```

File: bindings/c/tree-sitter-make.pc.in [hunk 4]
```diff
@@ -0,0 +1,11 @@
+prefix=@PREFIX@
+libdir=@LIBDIR@
+includedir=@INCLUDEDIR@
+
+Name: tree-sitter-make
+Description: Make grammar for tree-sitter
```
Comment (amaanq) [inline bindings/c/tree-sitter-make.pc.in:6]:
```suggestion
Description: Makefile grammar for tree-sitter
```
might be a little clearer

File: bindings/go/binding.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package tree_sitter_make
+
+// #cgo CFLAGS: -std=c11 -fPIC
+// #include "../../src/parser.c"
+// // NOTE: if your language has an external scanner, add it here.
```
Comment (amaanq) [inline bindings/go/binding.go:5]:
```suggestion
```

File: bindings/python/tree_sitter_make/__init__.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+"Make grammar for tree-sitter"
```
Comment (amaanq) [inline bindings/python/tree_sitter_make/__init__.py:1]:
```suggestion
"Makefile grammar for tree-sitter"
```

File: bindings/rust/build.rs [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    let parser_path = src_dir.join("parser.c");
+    c_config.file(&parser_path);
+    println!("cargo:rerun-if-changed={}", parser_path.to_str().unwrap());
+
+    // NOTE: if your language uses an external scanner, uncomment this block:
+    /*
+    let scanner_path = src_dir.join("scanner.c");
+    c_config.file(&scanner_path);
+    println!("cargo:rerun-if-changed={}", scanner_path.to_str().unwrap());
+    */
```
Comment (amaanq) [inline bindings/rust/build.rs:19]:
```suggestion
```

File: bindings/rust/lib.rs [hunk 8]
```diff
@@ -0,0 +1,54 @@
+//! This crate provides Make language support for the [tree-sitter][] parsing library.
+//!
+//! Typically, you will use the [language][language func] function to add this language to a
+//! tree-sitter [Parser][], and then use the parser to parse some code:
+//!
+//! ```
+//! let code = r#"
```
Comment (amaanq) [inline bindings/rust/lib.rs:7]: add 1-2 lines of some Make code so that this doc test succeeds

File: bindings/rust/lib.rs [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+///
+/// [`node-types.json`]: https://tree-sitter.github.io/tree-sitter/using-parsers#static-node-types
+pub const NODE_TYPES: &str = include_str!("../../src/node-types.json");
+
+// Uncomment these to include any queries that this grammar contains
+
+// pub const HIGHLIGHTS_QUERY: &str = include_str!("../../queries/highlights.scm");
+// pub const INJECTIONS_QUERY: &str = include_str!("../../queries/injections.scm");
+// pub const LOCALS_QUERY: &str = include_str!("../../queries/locals.scm");
+// pub const TAGS_QUERY: &str = include_str!("../../queries/tags.scm");
```
Comment (amaanq) [inline bindings/rust/lib.rs:43]:
```suggestion

pub const HIGHLIGHTS_QUERY: &str = include_str!("../../queries/highlights.scm");
```

File: package.json [hunk 10]
```diff
@@ -5,13 +5,11 @@
...
   "license": "MIT",
   "main": "bindings/node",
+  "types": "bindings/node",
   "homepage": "https://github.com/alemuller/tree-sitter-make#readme",
-
   "bugs": {
     "url": "https://github.com/alemuller/tree-sitter-make/issues"
   },
-  "homepage": "https://github.com/alemuller/tree-sitter-make#readme",
-
```
Comment (amaanq) [inline package.json]: Update urls here

File: pyproject.toml [hunk 11]
```diff
@@ -0,0 +1,29 @@
+[build-system]
+requires = ["setuptools>=42", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "tree-sitter-make"
+description = "Make grammar for tree-sitter"
```
Comment (amaanq) [inline pyproject.toml]:
```suggestion
description = "Makefile grammar for tree-sitter"
```

File: setup.py [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+        "tree_sitter_make.queries": ["*.scm"],
+    },
+    ext_package="tree_sitter_make",
+    ext_modules=[
+        Extension(
+            name="_binding",
+            sources=[
+                "bindings/python/tree_sitter_make/binding.c",
+                "src/parser.c",
+                # NOTE: if your language uses an external scanner, add it here.
```
Comment (amaanq) [inline setup.py:39]:
```suggestion
```

File: package.json [hunk 13]
```diff
@@ -5,13 +5,11 @@
   "author": "Alexandre Muller",
   "license": "MIT",
   "main": "bindings/node",
+  "types": "bindings/node",
   "homepage": "https://github.com/alemuller/tree-sitter-make#readme",
```
Comment (Freed-Wu) [inline package.json]:
```suggestion
  "homepage": "https://github.com/tree-sitter-grammars/tree-sitter-make#readme",
```

File: package.json [hunk 14]
```diff
@@ -5,13 +5,11 @@
...
-
   "bugs": {
     "url": "https://github.com/alemuller/tree-sitter-make/issues"
```
Comment (Freed-Wu) [inline package.json]:
```suggestion
    "url": "https://github.com/tree-sitter-grammars/tree-sitter-make/issues"
```

**Comment by domWalters** [comment]
@amaanq @Freed-Wu is there anymore work to be done before this can merge?

I'm a user of `tree-sitter-make` and would really like to be able to install it from PyPi, without having to use:
  - The very large [`tree-sitter-language-pack`](https://pypi.org/project/tree-sitter-language-pack/).
  - The unmaintained [`tree-sitter-languages`](https://pypi.org/project/tree-sitter-languages).
  - The yanked [`tree-sitter-make`](https://pypi.org/project/tree-sitter-make/0.0.1/)

**Comment by amaanq** [comment]
Superseded by #4

**Pull request closed without merging**